MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -2910,49 +3019,114 @@ InitializePageAttributesCallback (
   IN VOID       *Context
   )
 {
+  EFI_STATUS  Status;
+  EFI_EVENT   DisableNullDetectionEvent;
+
+  // Inidcates that the GCD sync process has been completed. This BOOLEAN
+  // must set this to TRUE so calls to ApplyMemoryProtectionPolicy() are not
+  // blocked. This BOOLEAN also allows guard pages to be set.
+  mGcdSyncComplete = TRUE;
+
+  // Initialize paging attributes
   InitializePageAttributesForMemoryProtectionPolicy ();
 
+  // Set all the guard pages
   HeapGuardCpuArchProtocolNotify ();
 
-  mPageAttributesInitialized = TRUE;
+  // Enable NULL pointer detection
+  EnableNullDetection ();

```

Comment:
This will unconditionally enable null detection, right? Is this no longer back by gDxeMps HOB?

Reply:
Good catch! Will update

Reply:
Fixed